src/scripts/urlUtils.ts

```diff
@@ -0,0 +1,143 @@
+import {ObjectUtils} from "./objectUtils";
+import {Settings} from "./settings";
+
+import {TooltipType} from "./clipperUI/tooltipType";
+
+export module UrlUtils {
+	export function checkIfUrlMatchesAContentType(url: string, tooltipTypes: TooltipType[]): TooltipType {
+		for (let i = 0; i < tooltipTypes.length; ++i) {
+			let tooltipType = tooltipTypes[i];
+			let contentTypeAsString = TooltipType[tooltipType];
+			let contentTypeRegexes = Settings.getSetting(contentTypeAsString + "Domains");
+			let concatenatedRegExes = new RegExp(contentTypeRegexes.join("|"), "i");
+			if (concatenatedRegExes.test(url)) {
+				return tooltipType;
+			}
+		}
+
+		return;
+	}
+
+	export function getFileNameFromUrl(url: string, fallbackResult?: string): string {
+		if (!url) {
+			return fallbackResult;
+		}
+		let regexResult = /\/(?=[^\/]+\.\w{3,4}$).+/g.exec(url);
+		return regexResult && regexResult[0] ? regexResult[0].slice(1) : fallbackResult;
+	}
+
+	export function getHostname(url: string): string {
+		let l = document.createElement("a");
+		l.href = url;
+		return l.protocol + "//" + l.host + "/";
+	}
+
+	export function getPathname(url: string): string {
+		let l = document.createElement("a");
+		l.href = url;
+
+		let urlPathName = l.pathname;
+
+		// We need to ensure the leading forward slash to make it consistant across all browsers.
+		return ensureLeadingForwardSlash(urlPathName);
+	}
+
+	function ensureLeadingForwardSlash(url: string): string {
+		url = ObjectUtils.isNullOrUndefined(url) ? "/" : url;
+		return (url.length > 0 && url.charAt(0) === "/") ? url : "/" + url;
+	}
+
+	/**
+	 * Gets the query value of the given url and key.
+	 *
+	 * @param url The URL to get the query value from
+	 * @param key The query key in the URL to get the query value from
+	 * @return Undefined if the key does not exist; "" if the key exists but has no matching
+	 * value; otherwise the query value
+	 */
+	export function getQueryValue(url: string, key: string): string {
```

Comment:
We should probably replace these methods with either [qs]<https://www.npmjs.com/package/qs> or [query-string]<https://www.npmjs.com/package/query-string>.